### DISCUSSION TITLE
Engineering Unit of Node

### DISCUSSION THREAD

**Discussion by bibhudatta-sahu90** [eclipse-milo/milo#925] [General]
Hello Everyone,

    I would like to know how can I get the "Engineering Unit" of a node while browsing the address space. Could you please help me on this topic.
    
Thanks,
Bibhudatta Sahu

**Comment by kevinherron** [comment]
EngineeringUnits are represented as an additional PropertyNode that instances of AnalogItemType may have.

If a Node has EngineeringUnits associated with it then when browsing that Node you would see a PropertyNode with a BrowseName of "EngineeringUnits".

**Reply by bibhudatta-sahu90** [reply]
Thanks Kevin. I am following below approach to get the Engineering Units.

client.getAddressSpace().getVariableNode(nodeId).readEngineeringUnitsAsync().get();

Is there any other approach to get Engineering Units (i.e from ReferenceDescription).

Thanks,
Bibhudatta Sahu

**Reply by kevinherron** [reply]
No they are not included in ReferenceDescription. You have to explicitly read them for any Node that may have them.

**Reply by bibhudatta-sahu90** [reply]
Thank you Kavin !!!
